src/Games/NexusMods.Games.RedEngine/Cyberpunk2077/Emitters/APathBasedDependencyEmitter.cs

```diff
@@ -0,0 +1,82 @@
+using System.Runtime.CompilerServices;
+using NexusMods.Abstractions.Diagnostics;
+using NexusMods.Abstractions.Diagnostics.Emitters;
+using NexusMods.Abstractions.Diagnostics.Values;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Extensions;
+using NexusMods.Paths;
+
+namespace NexusMods.Games.RedEngine.Cyberpunk2077.Emitters;
+
+/// <summary>
+/// This is an abstract class that represents an emitter for known dependencies that are based on paths. For example
+/// if you need to look for files of a given type and path, that require a mod loader which also has a specific path
+/// and file extension, this is what you need.
+///
+/// For example: Cyberpunk2077 has Cyber Engine Tweaks which installs a mod loader at the path `bin/x64/version.dll`
+/// and the mods are installed at `bin/x64/plugins/cyber_engine_tweaks/{ModName}/` with the file extension `.lua`.
+/// Armed just with these paths and this class, you can create a diagnostic that checks if the mod loader is installed
+/// and enabled.
+/// 
+/// </summary>
+public abstract class APathBasedDependencyEmitter : ILoadoutDiagnosticEmitter
+{
+    /// <summary>
+    /// The link to download the dependency.
+    /// </summary>
+    protected abstract NamedLink DownloadLink { get; }
+    
+    /// <summary>
+    /// The name of the dependency.
+    /// </summary>
+    protected abstract string DependencyName { get; }
+    
+    /// <summary>
+    /// All the paths returned by this property must exist for the dependency to be considered installed.
+    /// </summary>
+    protected internal abstract IEnumerable<GamePath> DependencyPaths { get; }
+    
+    /// <summary>
+    /// The folders that contain the dependant mods, for example `bin/x64/plugins/cyber_engine_tweaks`.
+    /// </summary>
+    protected internal abstract GamePath[] DependantPaths { get; }
+    
+    /// <summary>
+    /// The file extensions of the dependant files, for example `.lua` or `.dll`.
+    /// </summary>
+    protected internal abstract Extension[] DependantExtensions { get; }
+    
+    public async IAsyncEnumerable<Diagnostic> Diagnose(Loadout.ReadOnly loadout, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        // Cache the properties
```

Comment:
Method is marked `async` but doesn't await anything. Since async enumerables are special on their own, I'd just add this to make the compiler happy:

```csharp
await Task.Yield();
```